question22/main.js

```diff
@@ -0,0 +1,188 @@
+  wrapper.appendChild(table).appendChild(tbody).appendChild(getTableRowFragment(userData));
+  sortButtonClickEvent(userData);
+}
+
+const getTableHeadElement = () => {
+  const thead = createElementWithClassName("thead", "thead");
+  const tr = createElementWithClassName("tr", "tr");
+  const fragmentThElement = document.createDocumentFragment();
+  const userDataColumnKeys = Object.keys(userDataColumn);
+
+  userDataColumnKeys.forEach((key) => {
+    const th = createElementWithClassName("th", "th");
+    const value = userDataColumn[key].value;
+    th.textContent = value; 
+    userDataColumn[key].sort && th.appendChild(createSortButton());
+    fragmentThElement.appendChild(tr).appendChild(th); 
+  });
+
+  thead.appendChild(fragmentThElement);
+  return thead;
+}
+
+const getTableRowFragment = (userData) => {
+  const tableRowfragment = document.createDocumentFragment();
+  for (let i = 0; i < userData.length; i++) {
+    const tr = createElementWithClassName("tr", "tr");
+    tableRowfragment.appendChild(tr).appendChild(getTableDataFragment(userData[i]));
+  }
+  return tableRowfragment;
+}
+
+const getTableDataFragment = (userData) => {
+  const tableDatafragment = document.createDocumentFragment();
+  Object.keys(userDataColumn).forEach((key) => {
+    const td = createElementWithClassName("td", "td");
+    td.textContent = userData[key];
+    tableDatafragment.appendChild(td);
+  });
+  return tableDatafragment;
+};
+
+const createSortButton = () => {
+  const sortButton = createElementWithClassName("button", "sortButton");
+  sortButton.dataset.status = "default";
+  sortButton.id = "js-sortbtn";
+  return sortButton;
+}
+
+const sortButtonClickEvent = () => {
+  const tbody = document.querySelector('tbody');
+  const sortButtons = [...document.querySelectorAll(".sortButton")];
+  const trElement = [...document.querySelectorAll("tbody > tr")];
+  sortButtons.forEach((sortButton)=> {
+    sortButton.addEventListener('click', (e) => {
+      resetSortbuttonStatus(sortButtons, e.target);
+      const clickedCellIndex = e.target.parentElement.cellIndex;
+      const nextStatus = switchSortStatus(e.target);
+      sortButton.dataset.status = nextStatus;
+      const sortedRows = sortFunc(e.target,clickedCellIndex,trElement);
+      while (tbody.firstChild) {
+        tbody.removeChild(tbody.firstChild);
+    }
+      sortedRows.forEach((row) => {
+        tbody.appendChild(row);
+      });
+    });
+  })
+}
+
+const resetSortbuttonStatus = (sortButtons, target) => {
+  sortButtons.forEach((button)=> {
+    if (button !== target) {
+      button.dataset.status = Sort.Default;
+    }
+  })
+}
+
+const Sort = {
+  Default: "default",
+  Asc: "asc",
+  Desc: "desc"
+};
+
+const switchSortStatus = (target) => {
+  switch (target.dataset.status) {
+    case Sort.Default:
+      return Sort.Asc;
+    case Sort.Asc:
+      return Sort.Desc;
+    case Sort.Desc:
+      return Sort.Default; 
+    default:
+      return Sort.defaul;
```

Comment:
IMO
I was wondering if I could specify `default` movement as the default? I was wondering.
[sandbox](https://codesandbox.io/s/trusting-frog-eu3kbs?file=/main.js:4518-4682)

```js
  switch (target.dataset.status) {
    case Sort.Asc:
      return Sort.Desc;
    case Sort.Desc:
      return Sort.Default;
    default:
      return Sort.Asc;
  }
```

Reply:
Thank you for your review! I hadn't thought of this idea, but it is certainly true. The movement doesn't seem particularly problematic either, so I'll go with this idea. Thank you so much!!! 
(By the way, I got the "default" letters wrong again...! I am sorry about that..! )

fixed commit: https://github.com/csakyo/homework/pull/37/commits/5cf40fb80a0b4f99edd4339e2ba01fa01d5d6ff9